packages/bvaughn-architecture-demo/src/suspense/types.ts

```diff
@@ -23,7 +23,7 @@ export type Record<T> = PendingRecord<T> | ResolvedRecord<T> | RejectedRecord;
 // You can use a Promise for this, but Promises have a downside (the microtask queue).
 // You can also create your own "thennable" if you want to support synchronous resolution/rejection.
 export interface Thennable<T> {
-  then(onFulfill: (value: T) => any, onReject: () => any): void | Thennable<T>;
+  then(onFulfill: (value: T) => any, onReject?: (err: any) => any): void | Thennable<T>;
```

Comment:
Why did you make `onReject` optional? Can we keep it required? (**Edit**: Is it because of the change to the `isThennable` method?)

Also small nit: Abbreviating `error` doesn't buy us anything 😁  (but admittedly I dislike arbitrary abbreviations in code)

Reply:
`Promise.then()` doesn't require it and a `Promise` should be a `Thennable`. And yes, I noticed it when I changed `isThennable()` because then typescript complained about `createRecorder.ts` where `Thennable.then()` is only passed one argument.